File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加
     * @param tbBrand
     */
    public void add(TbBrand tbBrand);

    /**
     * 通过id查询单条记录
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);
    /**
     * 删除
     */
    public void delete(Long[] ids);

    /**
     * 模糊查询 分页
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand tbBrand,int pageNum, int pageSize);

    /**
     * 查询品牌下拉列表
     * @return
     */
    public List<Map> selectOptionList();

}

File: pinyougou_parent/pinyougou_Manager_web/src/main/webapp/js/service/brandService.js
//服务层
app.service("brandService",function ($http) {
    //查询所有品牌
    this.findAll=function () {
        return $http.get("../brand/findAll.do")
    }
    //分页查询
    this.findPage=function (page,rows) {
        return $http.get("../brand/findPage.do?page="+page+"&rows="+rows)
    }
    //添加操作
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity)
    }
    //更新操作
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity)
    }
    //查询单条记录
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id)
    }
    //批量删除
    this.delete=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids)
    }
    //模糊查询
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page="+page+"&rows="+rows,searchEntity)
    }
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do")
    }

})